analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextTest.Register.cs

```diff
@@ -200,6 +279,54 @@ private class DummyCodeBlockStartAnalysisContext<TSyntaxKind> : CodeBlockStartAn
             throw new NotImplementedException();
     }
 
+    private class DummyCompilationStartAnalysisContext : CompilationStartAnalysisContext
+    {
+        private readonly DummyAnalysisContext context;
+        private int compilationEndCount;
+        private int semanticModelCount;
+        private int symbolCount;
+        private int nodeCount;
+
+        public Diagnostic RaisedDiagnostic { get; private set; }
+
+        public DummyCompilationStartAnalysisContext(DummyAnalysisContext context) : base(context.Model.Compilation, context.Options, default)
+        {
+            this.context = context;
+        }
+
+        public void AssertExpectedInvocationCounts(int expectedCompilationEndCount = 0, int expectedSemanticModelCount = 0, int expectedSymbolCount = 0, int expectedNodeCount = 0)
+        {
+            compilationEndCount.Should().Be(expectedCompilationEndCount);
+            semanticModelCount.Should().Be(expectedSemanticModelCount);
+            symbolCount.Should().Be(expectedSymbolCount);
+            nodeCount.Should().Be(expectedNodeCount);
+        }
+
+        public override void RegisterCodeBlockAction(Action<CodeBlockAnalysisContext> action) => throw new NotImplementedException();
+        public override void RegisterCodeBlockStartAction<TLanguageKindEnum>(Action<CodeBlockStartAnalysisContext<TLanguageKindEnum>> action) => throw new NotImplementedException();
```

Comment:
EOLs to separate methods (as it's not a single block of single-lines anymore
```suggestion
        public override void RegisterCodeBlockAction(Action<CodeBlockAnalysisContext> action) => throw new NotImplementedException();
        
        public override void RegisterCodeBlockStartAction<TLanguageKindEnum>(Action<CodeBlockStartAnalysisContext<TLanguageKindEnum>> action) => throw new NotImplementedException();
        
```

Reply:
Also below